src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureIndex = random.nextInt(4);
+        switch (figureIndex) {
+            case 0:
+                return new Circle(colorSupplier.getRandomColor(), getRandomDouble());
+            case 1:
+                return new Square(colorSupplier.getRandomColor(), getRandomDouble());
+            case 2:
+                return new RightTriangle(colorSupplier.getRandomColor(),
+                        getRandomDouble(),
+                        getRandomDouble());
+            case 3:
+                return new Rectangle(colorSupplier.getRandomColor(),
+                        getRandomDouble(),
+                        getRandomDouble());
+            case 4:
+            default:
+                return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                        getRandomDouble(),
+                        getRandomDouble(),
+                        getRandomDouble());
```

Comment:
create separate methods for each figure that will be returning a random figure, for example, getRandomSquare() and so on